docs/start/framework/react/guide/selective-hydration.md

```
@@ -0,0 +1,475 @@
+The `hydrate` option is an **opt-in/opt-out mechanism**. Conflicts occur when:
+- Some routes explicitly set `hydrate: false` (opt-out)
+- Other routes explicitly set `hydrate: true` (opt-in to ensure hydration)
+
+**Note:** Routes that don't specify `hydrate` (using the default behavior) do not create conflicts.
+
+When TanStack Start detects conflicting explicit settings:
+
+1. **Does not hydrate the page** (safer default - respects the `false` setting)
+2. **Logs a warning** to help you debug:
+
+```
```

markdownlint-cli2 (0.21.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)